### PR TITLE
Basic Webhooks

### PR DESCRIPTION
**Pull request by dave-juicelabs**

### CODE REVIEW

File: cmd/controller/frontend/frontend.go [hunk 1]
```diff
@@ -75,8 +151,15 @@ func (frontend *Frontend) getAgentById(id string) (restapi.Agent, error) {
 	return frontend.storage.GetAgentById(id)
 }
 
-func (frontend *Frontend) updateAgent(update restapi.AgentUpdate) error {
-	return frontend.storage.UpdateAgent(update)
+func (frontend *Frontend) updateAgent(ctx context.Context, update restapi.AgentUpdate) error {
+	err := frontend.storage.UpdateAgent(update)
+	if err == nil && len(update.Sessions) > 0 {
+		for sessionId, session := range update.Sessions {
+			frontend.webhookMessages <- fmt.Sprintf("/v1/session/update?agentId=%s&sessionId=%s&newState=%s", update.Id, sessionId, session.State)
```
Comment (ykagan) [inline cmd/controller/frontend/frontend.go:158]: might need to URL Encode this?

Comment (ykagan) [reply cmd/controller/frontend/frontend.go:158]: or send the data as POST data instead of a string?

**Review by ykagan — APPROVED** [review]
lgtm

File: cmd/controller/frontend/frontend.go (hunk 1 above)
Comment (dave-juicelabs) [reply cmd/controller/frontend/frontend.go:158]: I was thinking of that as an alternative. The user provides a url, we post to that url with the session data. Seems cleaner?

**Pull request merged**